### PR TITLE
ARC: Add virtual uart over ARC hostlink channel support

### PR DESCRIPTION
**Pull request by evgeniy-paltsev**
Add support for virtual UART device that uses ARC Hostlink channels for data transfers. Due to the Hostlink principle, the driver    supports only polling API.

Hostlink is supported on nSIM and HW targets (if launched with MDB debugger).

### CODE REVIEW

File: drivers/serial/uart_hostlink.c [hunk 1]
```diff
@@ -0,0 +1,410 @@
...
+struct hl {
+	/* General hostlink information.  */
+	volatile struct hl_hdr hdr;
+	/* Start of the hostlink buffer.  */
+	volatile struct hl_pkt_hdr pkt_hdr;
+	/* Payload buffer */
+	volatile union payload_u payload;
+} __aligned(HL_MAX_DCACHE_LINE) __packed;
+
+/* In general we mast exact fit into one or multiple cache lines as we shouldn't share hostlink
```
Comment (ruuddw) [inline drivers/serial/uart_hostlink.c:120]: mast->must

File: drivers/serial/uart_hostlink.c [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+ * @param dev UART device struct
+ * @param c Pointer to character
+ *
+ * @return 0 if a character arrived, -1 if the input buffer if empty.
+ */
+static int uart_hostlink_poll_in(const struct device *dev, unsigned char *c)
+{
+	ARG_UNUSED(dev);
+
+	/* We plan to use hostlink for logging, so no much sense in poll_in implementation */
```
Comment (ruuddw) [inline drivers/serial/uart_hostlink.c:387]: If it is not too complex, why not add input? That enables shell functionality, and other simple interactivity if needed.

**Comment by ruuddw** [comment]
Great addition for bring-up and debugging. Maybe add a short description in the board documentation? Anything special commands required to start a debugger session that should be documented, like enable/disable hostlink in the debugger?

File: drivers/serial/uart_hostlink.c (hunk 2 above)
Comment (evgeniy-paltsev) [reply drivers/serial/uart_hostlink.c:387]: As I remember shell require interrupt functionality from UART, so implementing of polling is not enough. But I'll check if it is so.

File: drivers/serial/uart_hostlink.c (hunk 1 above)
Comment (evgeniy-paltsev) [reply drivers/serial/uart_hostlink.c]: fixed

**Comment by evgeniy-paltsev** [comment]
> Anything special commands required to start a debugger session that should be documented, like enable/disable hostlink in the debugger?

I've added hostlink-awareness to west runners - so no changes required if we start debugging session or just do with `mdb-hw`, `mdb-nsim`, `arc-nsim` west runners or with cmake scripting on nSIM.

**Review by abrodkin — APPROVED** [review]

File: drivers/serial/uart_hostlink.c (hunk 2 above)
Comment (ruuddw) [reply drivers/serial/uart_hostlink.c:387]: Agreed - hostlink stdin may be blocking instead of polling, let's continue with this first.

**Review by ruuddw — APPROVED** [review]

**Review by mbolivar-nordic — COMMENTED** [review]
Late to the party, but runner changes LGTM

**Pull request merged**